### DISCUSSION TITLE
util:RemoveFolderEx fails when folder Path/FileName is too long.

### DISCUSSION THREAD

**Discussion by mikehenselersimfront** [wixtoolset/issues#7599] [Questions] [answered]
Error 2318:  File does not exist:
When folder Path/FileName  is too long.

My folder depth is 14 folders deep.
Path/FileName 273 characters long

when i move the offending folder/files to a folder closer to the root(thus creating a shorter path) then RemoveFolderEx works as expected.

also:  RemoveFolderEx does not work when running from maintenance "REMOVE" uninstall, but does work when run from a full "UNINSTALL"(when not hitting error 2318)

Using Wix 3
Is this a known issue?
Is there a work around?

**Comment by barnson** [comment]
What's the actual log entry?

**Reply by mikehenselersimfront** [reply]
Action start 17:21:40: InstallValidate.
MSI (s) (30:14) [17:21:40:059]: Note: 1: 2205 2:  3: _RemoveFilePath 
MSI (s) (30:14) [17:21:40:878]: Note: 1: 2318 2: C:\Program Files\SimToolSuite\wringer-web\data\LDITProjectX\1.0.0-SNAPSHOT\features\LDITProjectX\Test Plans\ScaledUpTesting\3_VCCIs-SingleLayer_DynamicPositions\TEST_NoCore_DynamicComments\SingleLayer_DynamicPositions_-_Test_NoCore_DynamicComments@VCCI1.feature 
DEBUG: Error 2318:  File does not exist: C:\Program Files\SimToolSuite\wringer-web\data\LDITProjectX\1.0.0-SNAPSHOT\features\LDITProjectX\Test Plans\ScaledUpTesting\3_VCCIs-SingleLayer_DynamicPositions\TEST_NoCore_DynamicComments\SingleLayer_DynamicPositions_-_Test_NoCore_DynamicComments@VCCI1.feature

**Reply by barnson** [reply]
That means it's not a problem in the custom action; it's somewhere in MSI.

**Reply by mikehenselersimfront** [reply]
Note: 
These are runtime files, not installed or managed by MSI
 After moving those offending group of files up one folder, then RemoveFolderEx works as expected. 
thnks

**Comment by barnson** [comment, marked as answer]
Yeah, MSI probably can't handle >MAXPATH paths in general.

**Reply by mikehenselersimfront** [reply]
Possibly, since according to the RemoveFolderEx documentation, "The custom action that implements RemoveFolderEx does so by writing temporary rows to the RemoveFile table for each subfolder of the root folder you specify". 
So RemoveFolderEx is working with MSI tables and not just a brute force delete type action. 
I guess i will have to do my own brute force custom action to delete this folder.

thnks for your assistance.

**Reply by robmen** [reply]
Remember to handle rollback. That's why the custom action uses the Windows Installer to delete the files. It does the hard work of handling rollback.

**Comment by mikehenselersimfront** [comment]
Right, one of the reason i moved from NSIS to MSI.
more built in management.
thnks
